adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -354,6 +355,7 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts map[str
 					opI.ClusterName)
 				opID := opI.ClusterName
 				opI.ClusterName = getClusterName(responseInterceptClustersNamePrefix, organizationID, vHost, apiTitle, apiVersion, opID)
+				operationalRespInterceptorVal[method] = opI // since cluster name is updated
```

Comment:
also, can we add an int test to detect this issue?

Reply:
added tests in the PR https://github.com/wso2/product-microgateway/pull/2743